### ISSUE TITLE
Basic stuff...

### ISSUE THREAD

**Issue by Exulansilist** [inkle/ink#238]
Hello, I'm new in this community and I'm trying to figure out how does all this work.
Sincerly I just wanted to write a "story-game" much like a book-game or the once from inkle's company.
I started out with the inklewriter but there were some features missing from that I guess and I asked about it to inkle's company page. 
They pointed me to the ink or inky program and I was not even able to understand the difference, then I noticed there where other things like Unity or Github and I'm even more confused now...
I really wanna learn about this stuff but I don't even know if I'm asking for help about this in the right spot right now... 
Hope someone will be able to help me or to tell me where should I ask and I'm sorry for my english but I'm Italian so I hope everything will be quite understandable.

**Comment by y-lohse** [comment]
Hi!
Start by downloading Inky from [this page](https://github.com/inkle/inky/releases/tag/0.6.3). *Inky* is a tool to write these games, a bit like inklewriter.
Once you have Inky running, you'll need to get familiar with *ink*, which is the language used to write the game. [Here's the tutorial for that](https://github.com/inkle/ink/blob/master/Documentation/WritingWithInk.md). With both these things you should be good to go.
Hope it helps!

**Comment by Exulansilist** [comment]
Okay, I'll start with that, thanks a lot, but you'll probably hear from me again soon ahah
thanks again.

**Comment by Exulansilist** [comment]
So... I guess I'm really really noob with this stuff but...

What are exactly the difference between this inky download
https://github.com/inkle/inky
And this one...
https://github.com/inkle/inky/releases/tag/0.6.3

Because I'm not sure everything is working right... I'm doing just basic stuff to understand the basic mechanics of Inky but it seems it is not working properly and I'm just following the tutorial still...
It doesn't do the right things even if I copy and paste the tutorial examples and it gives me errors even if I open the .ink script from the intercept project demo...

I'm feeling so dumb... :(

**Comment by y-lohse** [comment]
> I'm feeling so dumb... :(

Yeah don't, there's no reason for that :-).

The two download links are the same. What doesn't work for you?
Here's a super simple script -- you can copy it in the left hand panel of inky, and it should work.

```
Hello world!
*   Hello back!
    Nice to hear from you!
-> END
```

If you get errors, you can copy them here -- maybe along with your script. It'll help figuring out what exactly is going wrong for you.

**Comment by Exulansilist** [comment]
First of all I noticed that I can start an Ink file with a knot or the preview on the right of inky does not start but the tutorial sais:

> Note that the game will automatically run the first knot it finds in a story if there is no "non-knot" content > 

 example of what I mean:
_== start == 
this is the start of the ink project
*first choice
    ->END
*second choice
    ->END_

I noticed I can fix this starting the file with I divert like this:
_->start_

But I noticed other people ink files don't have them as in their inky they don't need it...


Another problem was with the glue thing (<>) but I was not able to recreate the situation I was on, it just did not work and in the tutorial it seems like it's so easy to use the glue tool but I just had problems...

The last one was trying to run this script:
https://github.com/inkle/the-intercept/blob/master/Assets/Ink/TheIntercept.ink

At first it didn't show me any error but after I played it, in a couple of points it did just stop asking me if that was an end instead of giving to me the choices but this script is from the actual demo...

**Comment by hoverbird** [comment]
Hey @Exulansilist— if your story file doesn't begin with a knot or stitch, then the story runner will just start running the first written content it finds. That's probably what you're seeing in the examples that don't being with a divert.

I actually think for stories of any real complexity, you'll want to make sure everything is organized into at least a few big knots, each containing some stitches. So a better pattern is to do exactly what you figured out on your own: create a `=== Start` knot, and then divert to it at the top of your main story file. 

Can you post a code snippet of how you're using the glue syntax, how you expect it to work, and how it actually behaves? That'll help us figure out what the problem is.

**Comment by Exulansilist** [comment]
Yeah, what I was saying is that, in my case, if I do not divert to my first knot, my ink won't just run the first know automatically as the tutorial sais. 
Same with stiches: if I divert to a knot the first stich won't be runned if the knot does not start with a divert to it.

```
=== knot ===
=stich
bla bla bla
```
this does not work with me and I have to do this

```
->knot
===knot===
->stich
=stich
bla bla bla
```

for the glue thing I guess I may have figured it out since I'm not having troubles anymore but idk if I'll get other errors I'll sure let you know :)))

**Comment by hoverbird** [comment]
You can do the above (diverting to a stitch from outside of the knot that contains it) like this:

`-> knot.stitch`

I think they may have removed the "auto-run first knot" rule because it was confusing, but maybe haven't updated the docs yet? At any rate, I don't recommend relying on that behavior. Just start your story with a divert and you should be all good.

**Comment by Exulansilist** [comment]
Okay, thanks. And... I think I'm doing quite good with my script and I feel enough confident about it to show it to some of my friends (won't post it here cause it's in italian so... yeah...) but I don't know how am I really supposed to show it to them in a way they can play it.
I really want to learn how I can use unity with ink scripts but I haven't find any tutorial on how to do something elaborated as the intercept demo. 
Hopefully you know what kind of demo am I talking about... Actually I just started to understand how Unity works but still... I don't really know what to do. Is there any written or video tutorial about it?

**Comment by Exulansilist** [comment]
up

**Comment by y-lohse** [comment]
I'm afraid I can't really help you with the Unity part. I don't *think* there's a full tutorial on how to use Ink with Unity -- I'd say your best chance is to learn more about how Unity works in general (plenty of tutorials on that), and then using [this](https://github.com/inkle/ink/blob/master/Documentation/RunningYourInk.md) and the intercept demo, try to adapt things to your game.

Anyway, I just wanted to point out that if you want a quick way to show your script to friends, you can export it for the web from Inky. This will produce a playable version that can be opened in a browser, and you can just send your friends the files or host them on a webserver somewhere, if you have that.

**Comment by Exulansilist** [comment]
I tried exporting it for web and I get four files but I don't really know how to make it work. I open the index.html but nothing happend...

**Comment by hoverbird** [comment]
Maybe check out Blot: https://github.com/abroder/blot

It's super basic, but should give you a good starting point for how Unity's UI systems can interface with the Ink runtime. Where you take it from there is up to you!

**Comment by Exulansilist** [comment]
okay, thanks :)
